packages/query-core/src/hydration.ts

```diff
@@ -21,6 +21,7 @@ export interface DehydrateOptions {
 
 export interface HydrateOptions {
   defaultOptions?: {
+    deserialize?: (data: any) => any
```

Comment:
maybe `deserializePromise` or something would be better since this only runs on promises?